### PR TITLE
Move repoupdater observability definitions into internal package

### PR DESCRIPTION
**Pull request by efritz**
The rewrite of precise-code-intel services will need to add observability metrics to a lot of places. There are some definitions in the repoupdater package that would be useful here to be used in multiple places. This PR does the following:

**Edit:** Simplified diff.

### CODE REVIEW

File: internal/observability/metrics.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package observability
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// registerer exists so we can override it in tests
+var registerer = prometheus.DefaultRegisterer
```
Comment (mrnugget) [inline internal/observability/metrics.go:8]: This seems pretty lonely in this file. Why is the file called `metrics.go`? I'd move this to the file with the `MustRegister` function so it's not that distance from the call site.

Comment (efritz) [reply internal/observability/metrics.go:8]: It's used in both disk monitor and request meter. It's the only thing that wasn't taken out of this file.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -212,7 +212,7 @@ Dockerfile @sourcegraph/distribution
 /internal/rcache/ @keegancsmith
 /internal/redispool/ @keegancsmith
 /internal/store/ @keegancsmith
-/internal/metrics @keegancsmith @slimsag
+/internal/observability @keegancsmith @slimsag
```
Comment (slimsag) [inline .github/CODEOWNERS:215]: We really should not rename this package to observability, metrics is the perfect name for it. See https://github.com/sourcegraph/sourcegraph/issues/9773

Comment (efritz) [reply .github/CODEOWNERS:215]: Even if it includes logging (and possibly tracing utilities in the future)?

Comment (efritz) [reply .github/CODEOWNERS]: Reverted the name. Maybe we should create a new package for the ErrorLogger type then?

Comment (efritz) [reply .github/CODEOWNERS]: https://github.com/sourcegraph/sourcegraph/pull/10297/files#diff-50ad335c013a94f31f3e7ad5e8e9651dR1

Comment (slimsag) [reply .github/CODEOWNERS]: > Even if it includes logging (and possibly tracing utilities in the future)?

If that's the case, then you are likely right and I was wrong.

Comment (efritz) [reply .github/CODEOWNERS]: I'm going to suggest moving the log functionality into internal/logging for now and if we have enough cross-package use we can merge them later.

File: internal/metrics/operation.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+func (m *OperationMetrics) MustRegister(r prometheus.Registerer) {
+	r.MustRegister(m.Duration)
+	r.MustRegister(m.Count)
+	r.MustRegister(m.Errors)
+}
+
+// NewBasicOperationMetrics creates an OperationMetrics value without any
+// specific labels. The supplied operationName should be underscore_cased
+// as it is used in the metric name.
+func NewBasicOperationMetrics(subsystem, metricPrefix, operationName string) *OperationMetrics {
```
Comment (slimsag) [inline internal/metrics/operation.go:40]: I think you should just name this `NewOperationMetrics`

Comment (efritz) [reply internal/metrics/operation.go:40]: Done.

File: internal/metrics/operation.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
+package metrics
+
+import (
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// OperationMetrics contains three common metrics for any operation.
+type OperationMetrics struct {
```
Comment (unknwon) [inline internal/metrics/operation.go:10]: nit: 
```diff
-type OperationMetrics struct {
+type Operation struct {
```

Because it is now under `metrics` namespace.

Comment (efritz) [reply internal/metrics/operation.go:10]: Will update on cleanup pass.

**Pull request merged**